docs/generators/MSSqlDDL.md

```diff
@@ -0,0 +1,494 @@
+
+IF OBJECT_ID('ExampleSchemaExampleAudit','U') IS NOT NULL
+    DROP TABLE ExampleSchemaExampleAudit
+GO
+
+CREATE TABLE ExampleSchemaExampleAudit
+(
+  AuditId INTEGER IDENTITY(1,1) NOT NULL PRIMARY KEY
+, AuditAction INTEGER NOT NULL -- 1 = INSERT, 2 = DELETE, 3 = UPDATE
+, AuditWhen DATETIME NOT NULL
+, ID INTEGER NULL
+, SOME_DATA VARCHAR(100) NULL
+, USER_ID VARCHAR(50) NULL
+, TMSTAMP DATETIME NULL
+)
+GO
+
+IF OBJECT_ID('ExampleSchemaExampleAuditTrigger','TR') IS NOT NULL
+    DROP TRIGGER ExampleSchemaExampleAuditTrigger
+GO
+
+CREATE TRIGGER ExampleSchemaExampleAuditTrigger ON ExampleSchemaExample
+FOR INSERT, DELETE, UPDATE AS
+BEGIN
+  DECLARE @INSERT INT, @DELETE INT, @ACTION INT;
+  SELECT @INSERT = COUNT(*) FROM INSERTED;
+  SELECT @DELETE = COUNT(*) FROM DELETED;
+  IF @INSERT > 0 SELECT @ACTION = 1 ELSE SELECT @ACTION = 0;
+  IF @DELETE > 0 SELECT @ACTION = @ACTION + 2;
+  -- 1 = INSERT, 2 = DELETE, 3 = UPDATE
+  IF @ACTION = 2 BEGIN
+    INSERT INTO ExampleSchemaExampleAudit
+    SELECT @ACTION
+         , GETDATE()
+          , ID
+          , SOME_DATA
+          , USER_ID
+          , TMSTAMP
+    FROM DELETED;
+  END ELSE
+  BEGIN
+    INSERT INTO ExampleSchemaExampleAudit
+    SELECT @ACTION
+         , GETDATE()
+         , ID
+         , SOME_DATA
+         , USER_ID
+         , TMSTAMP
+    FROM INSERTED;
+  END
+END
+GO
+```
+
+
+## Index Information
+Indexes can be created using the `KEY` keyword in the SI file:
+
+Example SI file:
+```
+TABLE Example
+    ID                  SEQUENCE
+    SOME_DATA           CHAR(100)
+    USER_ID             USERSTAMP
+    TMSTAMP             TIMESTAMP
+
+KEY PKEY PRIMARY
+    ID
+
+KEY INX1
+    SOME_DATA
+
+KEY INX2
+    ID
+    SOME_DATA
+```
+Example output:
+```
+CREATE TABLE ExampleSchema.Example
+(
+  ID INTEGER IDENTITY(1,1) NOT NULL
+, SOME_DATA VARCHAR(100) NOT NULL
+, USER_ID VARCHAR(50) NOT NULL
+, TMSTAMP DATETIME NOT NULL
+, TIMESTAMP
+, CONSTRAINT PK_ExampleSchema_Example_PKEY PRIMARY KEY (
+    ID
+  )
+)
+GO
+
+CREATE INDEX INX1 ON ExampleSchema.Example
+(
+  SOME_DATA
+)
+GO
+
+CREATE INDEX INX2 ON ExampleSchema.Example
+(
+  ID
+, SOME_DATA
+)
+GO
+```
+## Unique constraints
+Unique constraits can be generated by adding the `UNIQUE` keyword after the name of the index.
+
+Example SI file:
+```
+TABLE Example
+    ID                  SEQUENCE
+    SOME_DATA           CHAR(100)
+    USER_ID             USERSTAMP
+    TMSTAMP             TIMESTAMP
+
+
+
+KEY PKEY PRIMARY
+    ID
+
+KEY INX1 UNIQUE
+    SOME_DATA
+```
+
+Example Output:
+```
+CREATE TABLE ExampleSchema.Example
+(
+  ID INTEGER IDENTITY(1,1) NOT NULL
+, SOME_DATA VARCHAR(100) NOT NULL
+, USER_ID VARCHAR(50) NOT NULL
+, TMSTAMP DATETIME NOT NULL
+, TIMESTAMP
+, CONSTRAINT PK_ExampleSchema_Example_PKEY PRIMARY KEY (
+    ID
+  )
+, CONSTRAINT UK_ExampleSchema_Example_INX1 UNIQUE (
+    SOME_DATA
+  )
+)
+GO
+```
+## Links
+Links can be created on fields to other table fields by using the `LINK` keyword:
```

Comment:
Consider renaming this section to "Foreign Keys", and adding a syntax explanation similar to KEY